internal/graph/graph.go

```diff
@@ -204,7 +205,8 @@ func (g *Graph) ConstructorArguments(ctype reflect.Type) ([]reflect.Value, error
 			}
 			args[idx] = v
 		} else {
-			return nil, fmt.Errorf("%v dependency of type %v is not registered", ctype, arg)
+			log.Printf("Assigning zero value for arguments in %v. Dependency of type %v is not yet registered", ctype, arg)
```

Comment:
Not sure we need logging here, but without the log message the responsibility of knowing if an object is of zero value shifts to users and forces them to call and log `dig.Validate()`